arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -383,31 +389,32 @@ static arangodb::Result cancelBarrier(network::ConnectionPool* pool,
   options.timeout = network::Timeout(timeout);
   options.skipScheduler = true; // hack to speed up future.get()
   
-  auto res =
+  auto response =
       network::sendRequest(pool, endpoint, fuerte::RestVerb::Delete,
                            REPL_BARRIER_API + std::to_string(barrierId),
                            VPackBuffer<uint8_t>(), options)
           .get();
 
-  if (res.ok()) {
-    auto* response = res.response.get();
-    if (response->statusCode() != fuerte::StatusOK &&
-        response->statusCode() != fuerte::StatusNoContent) {
-      std::string errorMessage = "got status " + std::to_string(response->statusCode());
-      LOG_TOPIC("f5733", ERR, Logger::MAINTENANCE)
+  auto res = response.combinedResult();
+
+  if (res.fail()) {
+    if (response.ok() && response.response->fail()) {
```

Comment:
this is more permissive now ?

Reply:
Yes, it is. But as far as I can tell, that's fine.